logstash-core/lib/logstash/agent.rb

```diff
@@ -34,6 +36,10 @@ def execute
     start_background_services
 
     @pipelines.each { |_, p| Thread.new { p.run } }
+    @web_agent = Thread.new do
+      LogStash::WebServer::Puma.new(@logger).run
+    end
+
```

Comment:
yeah, for now is broken, so I rather keep it here :cry:, but I do agree makes sense to be moved there.